traverson4j-hc5/src/main/java/uk/co/autotrader/traverson/http/ApacheHttpTraversonClientAdapter.java

```diff
@@ -37,32 +37,42 @@ public ApacheHttpTraversonClientAdapter(CloseableHttpClient client) {
 
     @Override
     public <T> Response<T> execute(Request request, Class<T> returnType) {
-        HttpUriRequest httpUriRequest = apacheHttpUriConverter.toRequest(request);
+        ClassicHttpRequest httpRequest = apacheHttpUriConverter.toRequest(request);
 
         HttpClientContext clientContext = HttpClientContext.create();
-        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
         AuthCache authCache = new BasicAuthCache();
 
         for (AuthCredential authCredential : request.getAuthCredentials()) {
-            UsernamePasswordCredentials userPassword = new UsernamePasswordCredentials(authCredential.getUsername(), authCredential.getPassword());
-            AuthScope scope = AuthScope.ANY;
+            UsernamePasswordCredentials userPassword = new UsernamePasswordCredentials(authCredential.getUsername(), authCredential.getPassword().toCharArray());
+            AuthScope authScope = new AuthScope(null, null, -1, null, null);
             if (authCredential.getHostname() != null) {
-                HttpHost target = HttpHost.create(authCredential.getHostname());
-                scope = new AuthScope(target);
+                HttpHost target;
+                try {
+                    target = HttpHost.create(authCredential.getHostname());
+                    authScope = new AuthScope(target);
+
+                } catch (URISyntaxException e) {
+                    throw new HttpException("Error with HttpHost", e);
+                }
+
                 if (authCredential.isPreemptiveAuthentication()) {
-                    authCache.put(target, new BasicScheme());
+                    BasicScheme authScheme = new BasicScheme();
+                    authScheme.initPreemptive(userPassword);
+                    authCache.put(target, authScheme);
                 }
             }
-            credentialsProvider.setCredentials(scope, userPassword);
+
+            credentialsProvider.setCredentials(authScope, userPassword);
         }
 
         clientContext.setCredentialsProvider(credentialsProvider);
         clientContext.setAuthCache(authCache);
         CloseableHttpResponse httpResponse = null;
         try {
-            httpResponse = adapterClient.execute(httpUriRequest, clientContext);
-            return apacheHttpUriConverter.toResponse(httpResponse, httpUriRequest.getURI(), returnType);
-        } catch (IOException e) {
+            httpResponse = adapterClient.execute(httpRequest, clientContext);
+            return apacheHttpUriConverter.toResponse(httpResponse, httpRequest.getUri(), returnType);
+        } catch (IOException | URISyntaxException e) {
```

Comment:
URISyntaxException, what did the v4 client used to do on this error scenario?

Reply:
It didnt throw because it was already a HttpUriRequest so I guess it was already deemed a valid URI? It's the `httpRequest.getUri()` method throwing the other exception type